assets/src/edit-story/components/canvas/element.js

```diff
@@ -0,0 +1,67 @@
+/**
+ * External dependencies
+ */
+import styled from 'styled-components';
+import PropTypes from 'prop-types';
+
+/**
+ * WordPress dependencies
+ */
+import { useLayoutEffect, useRef } from '@wordpress/element';
+
+/**
+ * Internal dependencies
+ */
+import { getDefinitionForType } from '../../elements';
+
+const Wrapper = styled.div``;
+
+function Element( {
+	isEditing,
+	addNodeForElement,
+	setClickHandler,
+	handleSelectElement,
+	element: {
+		id,
+		type,
+		...rest
+	},
+} ) {
+	const { Display, Edit } = getDefinitionForType( type );
+	const element = useRef();
+	const props = { ...rest, id };
+
+	useLayoutEffect( () => {
+		addNodeForElement( id, element.current );
```

Comment:
It's weird and pretty bad structure. It needs an overhaul when someone thinks of a better way to do it. Yes, it should probably be `setNodeForElement`. I'll rename it here, but this will probably also be completely changed when merged with @miina's stuff.

Reply:
I actually just commented on one of @miina's pull requests about the same. If we can set a DOM node on an element, things could get a lot simpler. Otherwise, we'll have to maintain some pretty long forwardRef chains to constantly try to find these mappings.